src/Lower.cpp

```diff
@@ -131,30 +143,31 @@ Module lower(const vector<Function> &output_funcs,
     // specializations' conditions
     simplify_specializations(env);
 
+    Stmt last_written;
+
     debug(1) << "Creating initial loop nests...\n";
     bool any_memoized = false;
     Stmt s = schedule_functions(outputs, fused_groups, env, t, any_memoized);
-    debug(2) << "Lowering after creating initial loop nests:\n"
-             << s << "\n";
+     debug(2) << "Lowering after creating initial loop nests:\n"
+             << new_or_empty(s, last_written) << "\n";
 
     if (any_memoized) {
         debug(1) << "Injecting memoization...\n";
         s = inject_memoization(s, env, pipeline_name, outputs);
         debug(2) << "Lowering after injecting memoization:\n"
-                 << s << "\n";
+                 << new_or_empty(s, last_written) << "\n";
     } else {
         debug(1) << "Skipping injecting memoization...\n";
     }
-
     debug(1) << "Injecting tracing...\n";
     s = inject_tracing(s, pipeline_name, trace_pipeline, env, outputs, t);
     debug(2) << "Lowering after injecting tracing:\n"
```

Comment:
This is awkward. Would it be better to have a little stateful class with a method:
```
LoweringLogger::operator()(const string &, const Stmt &);
...

log("Lowering after injecting parameter checks", s);
```